src/cluster.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import * as MapboxGL from 'mapbox-gl';
+import { Props as MarkerProps } from './marker';
+import supercluster from 'supercluster';
+import * as GeoJSON from 'geojson';
+
+export interface Props {
+  ClusterMarkerFactory(coordinates: number[], pointCount: number): JSX.Element;
+  clusterThreshold?: number;
+  radius?: number;
+  maxZoom?: number;
+  minZoom?: number;
+  extent?: number;
+  nodeSize?: number;
+  log?: boolean;
+  children?: Array<React.Component<MarkerProps, void>>;
+}
+
+export interface State {
+  superC: any;
+  clusterPoints: any[];
+}
+
+export interface Context {
+  map: MapboxGL.Map;
+}
+
+export interface Point {
+  type: string;
+  geometry: {
+    type: string;
+    coordinates: GeoJSON.Position;
+  };
+  properties: {
+    point_count: number;
+  };
+}
+
+export default class Cluster extends React.Component<Props, State> {
+  public context: Context;
+
+  public static contextTypes = {
+    map: React.PropTypes.object
+  };
+
+  public static defaultProps = {
```

Comment:
Pretty sure that you don't need to say that a static variable is `public`, since `private/protected static` shouldn't be a thing